packages/replay-next/src/utils/telemetry.ts

```diff
@@ -20,3 +31,16 @@ export async function recordData(event: string, tags?: Object): Promise<void> {
     }
   }
 }
+
+export function assertWithTelemetry(
+  condition: unknown,
+  event: string,
```

Comment:
What is `event` ? It seems significant but there's no explanation about what it is or how you would pick a meaningful name for it.

Reply:
It's the assert message. What would help here?

Reply:
>> What is `event` ? 
>
> It's the assert message.

Down below it looks like you're passing some other value:

```js
return assertWithTelemetry(value, "process-test-metadata", message, tags);
```

But I'm not sure what the significance of it is. I'm guessing it's some kind of unique type/enum (maybe that's why you chose event?)

> What would help here?

Maybe we could re-order the args to be closer tot he familiar `assert` pattern? Maybe something like this?

```ts
export function assertWithTelemetry(
  assertion: unknown,
  message: string,
  type: string,
  tags: Object = {}
): asserts value {
  if (!assertion) {
    recordData(type, tags);
  }

  return assert(assertion, message);
}
```

"Type" is also kind of vague.

Reply:
agreed. will clean up tonight

Reply:
Sure thing. Also feel free to hand off to me if you'd prefer.